src/corelibs/U2Core/src/datatype/msa/MultipleChromatogramAlignmentRow.cpp

```diff
@@ -400,56 +400,31 @@ QPair<DNAChromatogram::ChromatogramTraceAndValue, DNAChromatogram::ChromatogramT
     return {peaks[0], peaks[1]};
 }
 
-bool MultipleChromatogramAlignmentRowData::isRowContentEqual(const MultipleChromatogramAlignmentRow &row) const {
-    return isRowContentEqual(*row);
+bool MultipleChromatogramAlignmentRowData::isEqual(const MultipleAlignmentRowData &other) const {
+    CHECK(other.type == MultipleAlignmentDataType::MCA, false);
+    auto mcaRow = dynamic_cast<const MultipleChromatogramAlignmentRowData *>(&other);
+    SAFE_POINT(mcaRow != nullptr, "Not an MCA row!", false);
+    return isEqual(*mcaRow);
 }
 
-bool MultipleChromatogramAlignmentRowData::isRowContentEqual(const MultipleChromatogramAlignmentRowData &row) const {
-    if (MatchExactly == DNASequenceUtils::compare(sequence, row.getSequence()) && ChromatogramUtils::areEqual(chromatogram, row.chromatogram)) {
-        if (sequence.length() == 0) {
-            return true;
-        } else {
-            U2MsaRowGapModel firstRowGaps = gaps;
-            if (!firstRowGaps.isEmpty() && (U2Msa::GAP_CHAR == charAt(0))) {
-                firstRowGaps.removeFirst();
-            }
-
-            U2MsaRowGapModel secondRowGaps = row.getGapModel();
-            if (!secondRowGaps.isEmpty() && (U2Msa::GAP_CHAR == row.charAt(0))) {
-                secondRowGaps.removeFirst();
-            }
+bool MultipleChromatogramAlignmentRowData::isEqual(const MultipleChromatogramAlignmentRowData &other) const {
+    CHECK(this != &other, true);
+    CHECK(getName() == other.getName(), false);
+    return isEqualCore(other);
+}
 
-            if (firstRowGaps == secondRowGaps) {
-                return true;
-            }
-        }
+bool MultipleChromatogramAlignmentRowData::isEqualCore(const MultipleAlignmentRowData &other) const {
+    if (other.type == MultipleAlignmentDataType::MCA) {
+        auto mcaRow = dynamic_cast<const MultipleChromatogramAlignmentRowData *>(&other);
+        SAFE_POINT(mcaRow != nullptr, "Not an MCA row!", false);
```

Comment:
The same

Reply:
Same here: this is not a null-ptr error, this is an invalid type error.